### PR TITLE
[BEAM-4640] Samza runner postcommit ValidatesRunner job

### PR DESCRIPTION
**Pull request by xinyuiscool**
**Please** add a meaningful description for your change here

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Spark
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by xinyuiscool** [comment]
@kennknowles : please take a look. Thanks!

File: .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Samza.groovy [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+import common_job_properties
+import JobBuilder
+
+// This job runs the suite of ValidatesRunner tests against the Samza runner.
+JobBuilder.postCommitJob('beam_PostCommit_Java_ValidatesRunner_Samza_Gradle',
+  'Run Samza ValidatesRunner', 'Apache Samza Runner ValidatesRunner Tests', this) {
+  description('Runs the ValidatesRunner suite on the Samza runner.')
+  previousNames('beam_PostCommit_Java_ValidatesRunner_Samza')
```
Comment (kennknowles) [inline .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Samza.groovy]: These previous names are for when we rename a job, to keep the history. We don't need that, so you can just delete the lines.

**Comment by kennknowles** [comment]
I also noticed that `gradle :beam-runners-samza:validatesRunner` executes no tests. I looked around but couldn't figure out exactly why. Can you see about that?

File: .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Samza.groovy (hunk 1 above)
Comment (xinyuiscool) [reply .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Samza.groovy]: Fixed.

**Comment by xinyuiscool** [comment]
I just ran ./gradlew :beam-runners-samza:validatesRunner in my box and it ran 208 tests in the end. Is there anything you saw different in jenkins?

**Comment by kennknowles** [comment]
Really curious. When I run `./gradlew :beam-runners-samza:validatesRunner` on my box there are no tests run. Well anyhow having the Jenkins job is the right start.

**Pull request merged**